tests/tests/test_mtls.py

```diff
@@ -191,6 +259,40 @@ def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
         out = device.run("mender -show-artifact").strip()
         assert out == "mtls-artifact"
 
+    @MenderTesting.fast
+    @pytest.mark.parametrize("algorithm", ["rsa"])
+    def test_mtls_enterprise_hsm(self, setup_ent_mtls, algorithm):
+        # set up the mTLS test environment and return the device
+        device = self.common_test_mtls_enterprise(setup_ent_mtls, algorithm, True)
+
+        # prepare a test artifact
+        with tempfile.NamedTemporaryFile() as tf:
+            artifact = make_script_artifact(
+                "mtls-artifact", conftest.machine_name, tf.name
+            )
+            deploy.upload_image(artifact)
+
+        # deploy the update to the device
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert len(devices) == 1
+        deployment_id = deploy.trigger_deployment(
+            "mtls-test", artifact_name="mtls-artifact", devices=devices,
+        )
+
+        # now just wait for the update to succeed
+        deploy.check_expected_statistics(deployment_id, "success", 1)
+        deploy.check_expected_status("finished", deployment_id)
+
+        # verify the update was actually installed on the device
+        out = device.run(
+            "export SOFTHSM2_CONF=/softhsm/softhsm2.conf; mender -show-artifact"
```

Comment:
Why do we need `export SOFTHSM2_CONF=/softhsm/softhsm2.conf` here?

Reply:
this is one of the follow ups I mentioned, once mender is called it will print errors about the keys/certificates. I thought they were coming from the daemon, but it was this place. when using SoftHSM2 we have to have the `SOFTHSM2_CONF` in the env, otherwise it is impossible for the engine to locate the keys. the follow-up would be: what to do in case we have mtls enabled and we are calling -show-artifact, should we print the errors or even load the keys? (low priority I think for now)

Reply:
Thank you, I understand now. This is a good question, indeed we can discuss in a client team meeting if you want. But I don't think it is a big deal, this is quite an advance use case and we should assume the user knows enough of OpenSSL.